rust/src/lib/ieee8021x.rs

```diff
@@ -0,0 +1,29 @@
+use serde::{Deserialize, Serialize};
+
+use crate::NetworkState;
+
+#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Ieee8021XConfig {
```

Comment:
Please add the `non-exhaustive` as we do not want to include new structs without it.